jetty-servlet/src/main/java/org/eclipse/jetty/servlet/ServletContextHandler.java

```diff
@@ -1071,6 +1071,9 @@ private void checkDynamicName(String name)
         {
             if (isStarted())
                 throw new IllegalStateException();
+            
+            if (ServletContextHandler.this.getServletHandler().isInitialized())
```

Comment:
Is `initialized' really the right event?  It be set to true from calls to updateHandlers in mappings etc.
Isn't there a more explicit event that can be used?   Perhaps just have isDynamicAllowed() and set that explitictly to false at exactly the point that it is no longer allowed?

Reply:
There is no more explicit event that can be used. ServletHandler.initialize() is the method that is called that calls the init methods on the Servlets. Once we've passed into initialize() it should not be possible to use any of the dynamic methods - note that we're only preventing the servlets and filters from calling the dynamic methods here, as the listeners and servletcontainerinitializers will have already been called elsewhere (and are permitted to call the dynamic methods). Effectively ServletHandler.initialize() is really only called from ServletHandler.doStart().